src/huggingface_hub/snapshot_download.py

```diff
@@ -45,6 +80,14 @@ def snapshot_download(
         cache_dir, repo_id.replace("/", REPO_ID_SEPARATOR) + "." + model_info.sha
     )
 
+    if framework is not None:
+        wanted_file_list_names = FILE_LIST_NAMES + [framework_mapping[framework]]
+        model_info.siblings = [
+            file
+            for file in model_info.siblings
+            if file.rfilename in wanted_file_list_names
+        ]
```

Comment:
I think it would be cleaner to remove the files we don't want, rather to fetch the files we want. A repository should imo be able to evolve over time, with different files.

Also I'm having a hard time understanding how this interacts with other libraries that include the model hub, this looks very transformers-specific.

Reply:
That's what I am doing here. I am updating `model_info.siblings` to contain only the files needed for running. 

I agree with you that is very transformers-specific. I explained in the attached comment why I implemented this.